src/Farmer/Builders/Builders.Dns.fs

```diff
@@ -11,25 +11,39 @@ open DnsRecords
 type DnsZoneRecordConfig =
     { Name : ResourceName
       Type : DnsRecordType
-      TTL : int }
-    static member Create(name, ttl, recordType) =
+      TTL : int
+      Zone : LinkedResource option }
+    static member Create(name, ttl, zone, recordType) =
         { Name =
               if name = ResourceName.Empty then failwith "You must set a DNS zone name"
               name
           TTL =
               match ttl with
               | Some ttl -> ttl
               | None -> failwith "You must set a TTL"
+          Zone = zone
           Type = recordType }
-
-type CNameRecordProperties =  { Name: ResourceName; CName : string option; TTL: int option; TargetResource: ResourceName option }
-type ARecordProperties =  { Name: ResourceName; Ipv4Addresses : string list; TTL: int option; TargetResource: ResourceName option  }
-type AaaaRecordProperties =  { Name: ResourceName; Ipv6Addresses : string list; TTL: int option; TargetResource: ResourceName option }
-type NsRecordProperties =  { Name: ResourceName; NsdNames : string list; TTL: int option; }
-type PtrRecordProperties =  { Name: ResourceName; PtrdNames : string list; TTL: int option; }
-type TxtRecordProperties =  { Name: ResourceName; TxtValues : string list; TTL: int option; }
-type MxRecordProperties =  { Name: ResourceName; MxValues : {| Preference : int; Exchange : string |} list; TTL: int option; }
-type SrvRecordProperties =  { Name: ResourceName; SrvValues : SrvRecord list; TTL: int option; }
+    interface IBuilder with
+        member this.ResourceId = zones.resourceId this.Name
```

Comment:
Doesnt this need the parent zone when generated the resourceId?

Reply:
Also doesnt this depend on the record type which ResourceId function should be used?